ldi-orchestrator/ldio-application/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/repositories/PipelineFileRepository.java

```diff
@@ -0,0 +1,152 @@
+package be.vlaanderen.informatievlaanderen.ldes.ldio.repositories;
+
+import be.vlaanderen.informatievlaanderen.ldes.ldio.config.OrchestratorConfig;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.config.PipelineConfig;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.exception.PipelineAlreadyExistsException;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.exception.PipelineDoesNotExistsException;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.exception.PipelineParsingException;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.valueobjects.PipelineConfigTO;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static be.vlaanderen.informatievlaanderen.ldes.ldio.valueobjects.PipelineConfigTO.fromPipelineConfig;
+import static java.util.Optional.ofNullable;
+
+@Service
+public class PipelineFileRepository implements PipelineRepository {
+	public static final String EXTENSION_YML = ".yml";
+	public static final String EXTENSION_YAML = ".yaml";
+	private final Logger log = LoggerFactory.getLogger(PipelineFileRepository.class);
+	private final Map<String, SavedPipeline> activePipelines;
```

Comment:
This gave me a lot of confusion, as your initialisation first reads the files, then checks if it exists in `addPipeline` and throws an exception when it does. I was convinced this code was a problem until I actually booted everything and realised your PipelineFileRepository is secretly doing more than I expected. I feel like this repository should only read and write to disk. When I ask "exists", I expect to get "true" if the file is in the pipelines dir and "false" when it isn't.

This class has two "save" methods, one writing to file, another putting it in memory. It has two "find" methods, same story.. I feel like we are really violating the single responsibility principle here.